File: leet1/leet1.cpp
// https://leetcode.com/problems/two-sum/
#include <unordered_map>
#include <vector>

class Solution
{
public:
    std::vector<int> twoSum(std::vector<int> &nums, int target)
    {
        std::unordered_map<int, int> hashmap;
        for (int index = 0; index < nums.size(); index++)
        {
            hashmap[nums[index]] = index;
        }
        for (int index = 0; index < nums.size(); index++)
        {
            int aim = target - nums[index];
            if (hashmap.find(aim) != hashmap.end() && hashmap[aim] != index)
            {
                return {index, hashmap[aim]};
            }
        }
        return {};
    }
};

File: leet1/leet1.py
# https://leetcode.com/problems/two-sum/
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashmap = {}
        for index, num in enumerate(nums):
            hashmap[num] = index
        for index, num in enumerate(nums):
            aim = target - num
            if aim in hashmap and hashmap[aim] != index:
                return [index, hashmap[aim]]
        return []
